pyccel/codegen/printing/cwrappercode.py

```diff
@@ -0,0 +1,144 @@
+from pyccel.codegen.printing.ccode import CCodePrinter
+from pyccel.ast.core import Variable , Assign, FunctionDef
+from pyccel.ast.datatypes import NativeInteger, NativeReal, NativeComplex, NativeBool, NativeString
+from pyccel.ast.cwrapper import PyccelPyObject, PyArg_ParseTupleNode #PyBuildValueNode
+from pyccel.ast.type_inference import str_dtype
+from pyccel.ast.core import If, Nil, Return, FunctionCall , FunctionDef , PyccelNot
+from pyccel.errors.errors import Errors
+from pyccel.errors.messages import *
+from sympy.logic.boolalg      import Not
+
+errors = Errors()
+
+__all__ = ["CWrapperCodePrinter", "ccode"]
+
+pytype_registry = {
+        NativeInteger(): 'l',
+        NativeReal(): 'd',
+        NativeComplex():'c',
+        NativeBool():'p',
+        NativeString():'s', 
+        PyccelPyObject():'O'
+        }
+
+class CWrapperCodePrinter(CCodePrinter):
+    def __init__(self, parser, settings={}):
+        CCodePrinter.__init__(self, parser,settings)
+
+    def get_new_name(self, used_names, requested_name):
+        if requested_name not in used_names:
+            used_names.add(requested_name)
+            return requested_name
+        else:
+            return 'random_name'
+       #     return create_incremented_string(used_names, prefix=requested_name)
+
+    def get_PyArgParseType(self, dtype):
+        #TODO: Depends on type, rank, etc
+        if dtype is NativeBool():
+            return NativeInteger(), lambda arg, tmp: Assign(arg, Bool(tmp))
+        return dtype, None
+
+    def _print_FunctionDef(self, expr):
+        used_names = set([a.name for a in expr.arguments] + [r.name for r in expr.results])
```

Comment:
The function name should also be in the used names

Reply:
yeah thats true ill do it